elytron/ELY-2320-integrity-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,152 @@
+
+Filesystem Integrity is a planned upcoming change, but will not be added until after WildFly 26.1 is released
+
+The two approaches being considered are between signing each identity file, and just calculating the checksum of each identity file.
+
+The advantage of using just a straightforward checksum using a hash algorithm is that it will not be resource heavy, as an algorithm such as SHA256 or MD5 is very efficient, although it comes with a caveat that it is possible to produce collisions (false positivies) due to the lack of complexity of the algorithm. The disadvantage of using a simple checksum is that if an authorized user were to gain access to the master list of checksums that is used to verify each identity when accessed, they could change the checksum rendering that identity to be useless, until that checksum is amended. Is it also possible for the unauthorized user to modify the identity file, calculate the new checksum, and update the checksum list with an updated one. This would be very detrimental as we have no way of knowing that the file has been modified and is no longer valid.
+
+If the approach of signing a file was chosen, it would require an asymmetric key pair, a public key and private key. This would involve a key store and a method to ensure only the correct party has access to the private key.
+
+It could be made that the public key is available for everyone to use. This means anyone can decrypt the signature to get the hash of the identity, but only people with access to the key store for the realm would have access to the private key that encrypts the hash of each identity. This way an unauthorized user could view the hash via the public key, but cannot change the hash/signature of the identity.
+
+The approach that is taken is to use both options in conjunction. To do so a ``main_index`` file will be created, that contains the identities, and their file hashes (checksum). This ``main_index`` file will be used to verify the file validity before anything else. First the main_index hash will be verified with the current identity hash. If they are identical then it is safe to assume that the file is unchanged. But in the event that the hash's are not the same, then it will proceed with by verifying the signature. The only case where a file would be invalid is if both the hash's don't match, and the signature does not validate for the file. Otherwise the file integrity is intact.
+The details about adding a signature to an XML file is described here and also has the XSD schema: https://www.w3.org/TR/xmldsig-core2/#sec-xsdSchema
+
+If encryption is enabled for the realm, the contents of the ``main_index`` file will also be encrypted. Instead of looking like this
+[source]
+----
+firstUser.xml:d1b31f991807feb73dba25e075609950
+----
+it would look like this
+[source]
+----
+RUxZAUMQUO9rq4PRc3BB61B5oZSepATonjzjUriqbsy//5fw:RUxZAUMQ41IKUmpazIiXdB5oZSep4wQa67I+y35TLtssweZ
+----
+==== Hard Requirements
```

Comment:
Is it possible to enable this feature for an existing realm without a need of conversion? In case the encryption feature the `elytron-tool` has to be used.

Reply:
Yes this is doable, will add to Elytron and then update this document

Reply:
The process should be the opposite way - plan first, then implementation. It seems we don't need `elytron-tool` for this functionality as there is the `update-key-pair` CLI operation.

Reply:
Sorry will update this plan first. 
And that is correct we don't need the elytron tool feature, the CLI operation will work for this.